### ISSUE TITLE
Cannot reorder from the first try 

### ISSUE THREAD

**Issue by TomashKhamlai** [magento/magento2#15059]
### Preconditions
1. Latest Magento 2.3-develop

### Steps to reproduce
1. Create a Virtual Product
2. Register a Customer Account
3. Place an Order with the Virtual Product
4. Create an Invoice
5. Try to Reorder

### Expected result
 ![exp](https://user-images.githubusercontent.com/11827230/39755389-53b9f57e-52ce-11e8-88bd-e1d9aeba74f4.png)

![exp1](https://user-images.githubusercontent.com/11827230/39755426-735329b4-52ce-11e8-835d-a2f828fc2cae.png)

### Actual result 
![act1](https://user-images.githubusercontent.com/11827230/39755415-6a7cce26-52ce-11e8-87d6-825b0ef8cc98.png)

![act](https://user-images.githubusercontent.com/11827230/39755434-7be6dee0-52ce-11e8-8f96-0b052e9c8ff9.png)

**Comment by engcom-backlog-nickolas** [comment]
Hello @TomashKhamlai, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by arnoudhgz** [comment]
@TomashKhamlai the re-order part, you did this from the back-end, I assume? 

I noticed that after clicking the reorder button the items in the cart are not complete, amount is set, but no price and subtotal....

![image](https://user-images.githubusercontent.com/10864061/44003661-9f162f46-9e56-11e8-90a7-6d29c16331af.png)

If you do a refresh of the page these values are filled normal and you are able to submit the order.

I can confirm this only is an issue with Virtual Products

**Comment by TomashKhamlai** [comment]
@arnoudhgz, yes, I did it from the back-end.
The problem with zero prices is also present in the checkout when Customer adds products to the Cart using Web API.

```
# Uncomment and modify
# endpoint="http://magento.loc/rest"
# username="admin"
# password="123123q"
# customer_email="your@email.here"

# Get Admin Token
admin_token=$(curl -X POST "$endpoint/V1/integration/admin/token" \
--header "Content-Type: application/json" \
-d '{"username":"'"$username"'","password":"'"$password"'"}') && echo $admin_token && admin_token=$(echo $admin_token | tr -d '"')

#Create 'Category 1' navigation link
curl -X POST "$endpoint/V1/categories" \
-H "Content-Type: application/json" \
-H "Authorization: Bearer $admin_token" \
-d '{"category":{"parent_id":2,"name":"Category 1","is_active":true,"level":2,"path":"1/2/3","available_sort_by":[],"include_in_menu":true,"custom_attributes":[{"attribute_code":"is_anchor","value":"1"},{"attribute_code":"path","value":"1/2/3"},{"attribute_code":"url_key","value":"category-1"},{"attribute_code":"url_path","value":"category-1"}]}}' | json_pp

#Create Simple Product
category_id="3"; productSKU1="Simple Product 1" && curl -X POST "$endpoint/V1/products" \
-H "Content-Type: application/json" \
-H "Authorization: Bearer $admin_token" \
-d '{"product":{"sku":"'"$productSKU1"'","name":"'"$productSKU1"'","attribute_set_id":4,"price":10.99,"status":1,"visibility":4,"type_id":"simple","created_at":"2018-01-01 00:00:01","updated_at":"2018-01-01 00:01:00","weight":1,"extension_attributes":{"category_links":[{"position":0,"category_id":"3"}],"stock_item":{"qty":1000,"is_in_stock":true,"is_qty_decimal":false}}}}' | json_pp

#Create a Customer
curl -X POST $endpoint/V1/customers \
-H "Content-Type: application/json" \
-H "Authorization: Bearer $admin_token" \
-d '{"customer":{"default_billing":"1","default_shipping":"1","email":"'"$customer_email"'","firstname":"Yokoshima","lastname":"Hiteca","store_id":1,"website_id":1,"addresses":[{"id":1,"region":{"region_code":"CA","region":"California","region_id":12},"region_id":12,"country_id":"US","street":["6161 West Centinela Avenue"],"company":"Magento","telephone":"555-55-555-55","postcode":"90230","city":"Culver City","firstname":"Yokoshima","lastname":"Hiteca","default_shipping":true,"default_billing":true}],"disable_auto_group_change":0,"extension_attributes":{"is_subscribed":false}},"password":"SuperPassword1"}' | json_pp

#Get the Customer Token
customer_token=$(curl -X POST "$endpoint/V1/integration/customer/token" \
 -H "Content-Type: application/json" \
 -d '{"username":"yhiteca@gmail.com","password":"SuperPassword1"}') && echo $customer_token && customer_token=$(echo $customer_token | tr -d '"')

#Get your Card Id
cart_id=$(curl -X POST "$endpoint/V1/carts/mine" \
-H "Content-Type: application/json" \
-H "Authorization: Bearer $customer_token") && echo $cart_id && cart_id=$(echo $cart_id | tr -d '"')

#Add the product to the Cart
curl -X POST "$endpoint/V1/carts/mine/items" \
-H "Content-Type: application/json" \
-H "Authorization: Bearer $customer_token" \
-d '{"cartItem":{"sku":"'"$productSKU1"'","qty":1,"quote_id":"'"$cart_id"'"}}'
```
Login as customer "username":"your@email.here","password":"SuperPassword1" and check it out.

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by shikhamis11** [comment]
I am working on this issue at #MM19IN

**Comment by shikhamis11** [comment]
@okorshenko I have also worked on this issue and proposed solution in my pull request . it is already assigned to @sivaschenko you can also check

**Comment by magento-engcom-team** [comment]
Hi @TomashKhamlai. Thank you for your report. 
The issue has been fixed in magento/magento2#21335 by @shikhamis11 in 2.3-develop branch
Related commit(s):
 - [3e832846dac8640fed86f8e0f0984f027c2f8b04](https://github.com/magento/magento2/commit/3e832846dac8640fed86f8e0f0984f027c2f8b04)
 - [124b0e03bd3004b04f2efe3ecac4d758fc4891c5](https://github.com/magento/magento2/commit/124b0e03bd3004b04f2efe3ecac4d758fc4891c5)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @TomashKhamlai. Thank you for your report. 
The issue has been fixed in magento/magento2#21513 by @shikhamis11 in 2.2-develop branch
Related commit(s):
 - [8493e59539df034b78b3424c37c3c87c927c724a](https://github.com/magento/magento2/commit/8493e59539df034b78b3424c37c3c87c927c724a)
 - [65e9346166084c859b0e1d58e8448aabc9f4bc49](https://github.com/magento/magento2/commit/65e9346166084c859b0e1d58e8448aabc9f4bc49)

The fix will be available with the upcoming 2.2.9 release.
